nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -480,9 +480,24 @@ def fwd_output_and_loss_func(batch, model, checkpoint_activations_all_layers=Non
             )
 
             def loss_func(output_tensor):
-                loss = self.loss_func(loss_mask, output_tensor)
-                reduced_loss = average_losses_across_data_parallel_group([loss])
-                return loss, {'avg': reduced_loss}
+                loss_for_mb = self.loss_func(loss_mask, output_tensor)
+                if validation_step and not self.cfg.data.get('validation_drop_last', True):
+                    num_valid_samples_in_mb = int(loss_mask.sum() / loss_mask.numel() * loss_mask.shape[0])
+                    loss_sum_for_mb = num_valid_samples_in_mb * loss_for_mb
```

Comment:
Can loss_sum_for_mb be calculated as loss_for_mb * loss_mask.sum()? Or is it missing something?